learningequality/kolibri
# Hide in-progress quiz scores in coach reports

### Summary
This change addresses issue https://github.com/learningequality/kolibri/issues/5585

Two items are described there, only one was found to be an issue:
1. "Started" - Any non-completed quiz scores were being shown in Coach Quiz Reports, so a student who hasn't "Submitted" their quiz might show a 50% when they've only gone through 5/10 questions and have gotten them all right.

2. Ensure that the "Average Score" at the top wasn't taking non-completed scores into its calculations. I did not find that this was happening so no change in this PR affects that calculation.

**NOTE** - This was initially targeted for 0.12.4 but because of additional complications after a Slack discussion, this PR will update the way the `Progress` column's data is made. Anybody who has "Started" their quiz, should have an indicator saying "X of Y Answered" rather than just "Started".

This will require a change to a serializer in class_summary_api.py and associated tests to make the necessary data available to the frontend.

### Reviewer guidance


1. Create a Class with Learners and an active Quiz assigned to that Class.
2. In one browser, log in as a coach and view the report for that quiz, listing all Learners and their respective statuses on that quiz.
3. In a separate browser, log in as one of the Learners (in Incognito / Private Browsing mode).
4. Open the quiz: _See that student's record show "X of Y Started" in the "Progress" column and that there is nothing in the score column.
5. Answer questions (some correctly) - _Note that nothing changes for that user and no score is shown_ Also note that if you view the report simultaneously, the X of Y will update as the learner answers questions.
6. Submit the Quiz as the Learner. _After a second, it will update automatically and show "Completed" and a value % in in the 'Score' column.

Other Learner scenarios to test:
- Submit a quiz with no questions answered
- Submit an incomplete quiz with _no correct_ answers given.
- Submit an incomplete quiz with correct answers given.
- Submit a fully completed quiz with correct answers given.

Also - note that the Average score will only change once you've submitted a quiz as a learner.

### References

https://github.com/learningequality/kolibri/issues/5585

### Contributor Checklist


PR process:


Testing:


### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md
